pkg/controller/deployment/deployment_controller.go

```diff
@@ -438,41 +438,124 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	if d.Spec.Paused {
-		// TODO: Implement scaling for paused deployments.
-		// Don't take any action for paused deployment.
-		// But keep the status up-to-date.
-		// Ignore paused deployments
-		glog.V(4).Infof("Updating status only for paused deployment %s/%s", d.Namespace, d.Name)
-		return dc.syncPausedDeploymentStatus(d)
+		return dc.sync(d)
 	}
+
 	if d.Spec.RollbackTo != nil {
 		revision := d.Spec.RollbackTo.Revision
 		if _, err = dc.rollback(d, &revision); err != nil {
 			return err
 		}
 	}
 
+	if dc.isScalingEvent(d) {
+		return dc.sync(d)
```

Comment:
Why should the deployment be re-queued after each sync?

On Tue, May 24, 2016 at 6:33 PM, Phillip Wittrock <notifications@github.com>
wrote:

> In pkg/controller/deployment/deployment_controller.go
> <https://github.com/kubernetes/kubernetes/pull/20273#discussion_r64427267>
> :
>
> >  	if d.Spec.RollbackTo != nil {
> >  		revision := d.Spec.RollbackTo.Revision
> >  		if _, err = dc.rollback(d, &revision); err != nil {
> >  			return err
> >  		}
> >  	}
> >
> > +	if dc.isScalingEvent(d) {
> > +		return dc.sync(d)
>
> Agreed that we want to requeue on RS events, but this code is relying upon
> that side affect. sync() doesn't provide a contract that it will re-queue
> the RS, it just happens to as an implementation detail, which is something
> we shouldn't have to know about if just reading this part of the code. If
> the code structure was altered so that sync could potentially not update
> any RS, then the deployment would not get re-queued, and our assumptions
> would break. If we want to rely upon that behavior, it should be part of
> the function's contract.
>
> Basically at this point in the code I want to be able to reason that the
> deployment will always be re-queued after the sync without necessarily
> knowing the implementation details of how or why.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20273/files/288d0835dbcd37ad101db417cf8e4395713ea64a#r64427267>
>